sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -106,22 +106,46 @@ private[sql] class RocksDBStateStoreProvider
     override def metrics: StateStoreMetrics = {
       val rocksDBMetrics = rocksDB.metrics
       def commitLatencyMs(typ: String): Long = rocksDBMetrics.lastCommitLatencyMs.getOrElse(typ, 0L)
-      def avgNativeOpsLatencyMs(typ: String): Long = {
-        rocksDBMetrics.nativeOpsLatencyMicros.get(typ).map(_.avg).getOrElse(0.0).toLong
+      def nativeOpsLatencyMillis(typ: String): Long = {
+        rocksDBMetrics.nativeOpsMetrics.get(typ).map(_ * 1000).getOrElse(0)
```

Comment:
Is it `nativeOpsHistograms` here?

Reply:
its actually `nativeOpsMetrics` (the ticker based metrics)

Reply:
Hm, `nativeOpsMetrics` looks like counting metrics, why this is `nativeOpsLatencyMillis` which is a timing metrics?

Reply:
The ticker based metrics are not just `counts` of some operations, they also capture the time. Example one is [here](https://javadoc.io/static/org.rocksdb/rocksdbjni/6.4.6/org/rocksdb/TickerType.html#STALL_MICROS). This is just a utility method to extract the time out of these ticker based metrics.